From: Ben Hutchings <benh@debian.org>
Date: Sat, 15 Jan 2022 22:30:49 +0100
Subject: libapi: Define _FORTIFY_SOURCE as 2, not empty

Signed-off-by: Ben Hutchings <benh@debian.org>
---
--- a/tools/lib/api/Makefile
+++ b/tools/lib/api/Makefile
@@ -29,7 +29,7 @@ endif
 endif
 
 ifeq ($(DEBUG),0)
-  CFLAGS += -D_FORTIFY_SOURCE
+  CFLAGS += -D_FORTIFY_SOURCE=2
 endif
 
 # Treat warnings as errors unless directed not to
